### DISCUSSION TITLE
How to maintain added image position when zoom out  in richitemcontrol

### DISCUSSION THREAD

**Discussion by praseethacp** [mircea21S/RichCanvas#37] [Q&A]
Hi i have added an item to the RichItemControl . I added that image with the same height and width as Viewport. I need to dock the image position in the container that means i should able to zoom in the image but the zoom out limit should be the 100 % of the image with keeping their initial position.Now when i try to zoom in and zoom out, the images position will vary . Is there any way to do this ??

Thank You !

**Comment by mircea21S** [comment]
Hi!

What you've described is not that easy to achieve. What you can try is to set the MaxScale limit to 1 so you have the initial viewport of the image when zooming out.

You can try to listen to the Zooming event on RichItemsControl and call BringIntoView on that item with the image, as right now zooming is based on mouse position (this might be a new feature to be able to choose how the zooming behaves). Calling BringIntoView should put the item in the center of the viewport.

Otherwise, you will have to wait after the next version release so I can give it a try maybe there are some limitations from the library and I might offer some features. 

That's all I can say for now. Hope it helps!

**Comment by praseethacp** [comment]
Thanks for the reply.
Could you help me with another doubt, i'm in a trouble right now !!!!!!
I need to zoom in and zoom out the main image with the scale 100,150.200 and out the same without the mouse event.  I binded Scale property of richitemcontrol so the image getting zoomed but the position is not maintaining.Is there any property to manage this ????

Simply saying i have plus icon and minus icon like map, when i tried to click plus sign the image should be zoom 100% then 150%. and when i click minus button it should zoom out . zoom out limit should be the first position of main image. 
Thanks in advance!!!!

**Reply by mircea21S** [reply]
Hey!

Firstly, I want to apologize for missing this comment. Either I've seen it and forgot about it or I didn't get notified about it and I also didn't check on this discussion thread. Sorry!

To answer your last comment if you want to use plus and minus signs to zoom in or out using your values, moreover to keep your image in place you should use RichItemsControl.ScrollContainer.Zoom() here you should pass the center of image position relative to the canvas and a delta which should be greater than zero for zoom in or less than zero for zoom out. 
I don't know what you are using now, but this method should zoom while keeping your image in place.
Regarding, zoom out limit you should set MinScale property for that. Probably 1 is that, I don't know.

This answer relates to the current released version v2.4.2. Let me know if this works for you or if you have any other questions! Hope it helps!
